### PR TITLE
Testthat

### PR DESCRIPTION
**Pull request by meta00**
This PR includes:
- tests added, thus resolved issue #8 
- modification to remove various redundancies and resolved issue #20

### CODE REVIEW

File: R/experiment.R [hunk 1]
```diff
@@ -36,46 +41,77 @@ new_experiment <- function(parameters, obsrates, tmax, seed,
...
+#    } else if (!is.null(parameters)) {
+#        dic_g2r <- setNames(paste0("p_", dic_g2r[sel1]), names(dic_g2r[sel1]))
+#        dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    } else if (!is.null(obsrates)) {
+#        dic_g2r <-  setNames(paste0("r_", dic_g2r[sel2]), names(dic_g2r[sel2]))
+#        dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    }
+#  } else {
+#     dic_g2r <- NULL
+#      dic_r2g <- NULL
```
Comment (LucieContamin) [inline R/experiment.R]: If these lines are on comment maybe we can delete them

File: R/list_experiments.R [hunk 2]
```diff
@@ -25,8 +25,6 @@ list_experiments <- function(file){
     if (grepl("type:", exp)) {
       experiment <- trimws(substr(exp, 1, regexpr("type:", exp) - 1))
       type <- trimws(substr(exp, regexpr("type:", exp) + 5, nchar(exp)))
-      if (grepl("\\s+", type))
-        type <- trimws(substr(type, 1, regexpr("\\s+", type)))
```
Comment (LucieContamin) [inline R/list_experiments.R:29]: why we don't need that anymore?

File: R/plot_params.R [hunk 3]
```diff
@@ -22,31 +22,38 @@
...
 
   # check n the number of parameters to be plotted
   exp <- as.data.frame(exp)
-  # if n is equal to 0
-  if (n == 0) stop("There is no parameters to plot in this experiment")
-  if (n == 1) stripchart(exp[, topidx[1]], xlab = colnames(exp)[topidx[1]])
-  if (n == 2) scatter2D(exp[, topidx[1]], exp[, topidx[2]],
+  # n can't be 0 once length(allvar) !=0
+#  if (n == 0)
+#    stop("There is no parameter to plot in this experiment")
```
Comment (LucieContamin) [inline R/plot_params.R]: why we don't need that anymore?

**Review by LucieContamin — COMMENTED** [review]
Perfect, thank you very much!

File: R/experiment.R (hunk 1 above)
Comment (meta00) [reply R/experiment.R]: ok

File: R/list_experiments.R (hunk 2 above)
Comment (meta00) [reply R/list_experiments.R:29]: because we use trimws function to produce 'type' already.

File: R/plot_params.R (hunk 3 above)
Comment (meta00) [reply R/plot_params.R]: i think these lines are repetitive. I reorganised the flow.

**Pull request merged**